### ISSUE TITLE
Not all modules processed by plugin

### ISSUE THREAD

**Issue by dschewchenko** [Menci/vite-plugin-top-level-await#21]
Hello. We have an MF setup. In one remote module we have exposed file with imports where we have dependency on shared module ("vue"). After transformation it parses only one imported module but not fully, in exposed module.
Sorry, because of NDA I cannot share full code.
Here 3 screenshots, all from one exposed module. No error/warning during transformation. build.target not set

Exposed module
<img width="853" alt="Знімок екрана 2023-01-16 о 18 31 04" src="https://user-images.githubusercontent.com/8172940/212737477-50c16443-9616-407f-bc52-11eca867a89f.png">

Transformed module, but not await importShared("vue")
<img width="681" alt="Знімок екрана 2023-01-16 о 18 31 23" src="https://user-images.githubusercontent.com/8172940/212737481-cdf00b0d-7b55-42f9-81c3-cb14d342b003.png">

Not transformed:
<img width="809" alt="Знімок екрана 2023-01-16 о 18 30 51" src="https://user-images.githubusercontent.com/8172940/212737473-599008b3-5fa3-4863-a322-a44597f60211.png">

**Comment by ds-nemesis** [comment]
Note: using https://github.com/originjs/vite-plugin-federation for MF. Changing order of plugins do nothing

**Comment by dschewchenko** [comment]
Note: using https://github.com/originjs/vite-plugin-federation for MF. Changing order of plugins do nothing

**Comment by dschewchenko** [comment]
on second screen joined mixins declaration, there is no top-level await, and your plugin wraps it in async function

**Comment by dschewchenko** [comment]
Downgrade to vite 3 do nothing. Same strange transformation

**Comment by Menci** [comment]
A reproduce project?

**Comment by dschewchenko** [comment]
Hello. Sorry, I have no time for creating reproduction repo, because of hard work and really need fix for this build for production. But can give you link for one https://github.com/originjs/vite-plugin-federation/tree/main/packages/examples/vue3-demo-esm

**Comment by Menci** [comment]
<img width="768" alt="image" src="https://user-images.githubusercontent.com/11341955/215116747-0fbfc29c-8735-47b3-b88f-cd03cd257122.png">

I can't install your example project locally.

And it seems to have no top-level awaits in its code (although it included this plugin in Vite)

**Comment by Menci** [comment]
Oh, I missed. It's pnpm

**Comment by dschewchenko** [comment]
workspace:* change to last available version and it will install. if npm:)
Screens of what generated I have sent in first message. await statements are generated. After that

**Comment by Menci** [comment]
The problem is that your plugin that injects `await importShared(...)` runs ***after*** my plugin's transform. I think adding `enforce: "post"` will fix it:

```js
import { importShared, __tla as __tla_0 } from "./__federation_fn_import.js";
import { createApp } from "./runtime-dom.esm-bundler-27e026b6.js";
import Content, { __tla as __tla_1 } from "./__federation_expose_Content-d811405f.js";
import button, { __tla as __tla_2 } from "./__federation_expose_Button-5cacfba7.js";
import UnusedButton from "./__federation_expose_UnusedButton-3784640f.js";
import Images from "./__federation_expose_Images-091d7789.js";
import { _export_sfc } from "./_plugin-vue_export-helper-e9a2c33e.js";
import { ref, resolveComponent, openBlock, createElementBlock, createVNode, pushScopeId, popScopeId, createBaseVNode } from "./runtime-core.esm-bundler-242c5926.js";
Promise.all([
  (() => {
    try {
      return __tla_0;
    } catch {
    }
  })(),
  (() => {
    try {
      return __tla_1;
    } catch {
    }
  })(),
  (() => {
    try {
      return __tla_2;
    } catch {
    }
  })()
]).then(async () => {
  const { createPinia } = await importShared("pinia");
  (function polyfill() {
    const relList = document.createElement("link").relList;
    if (relList && relList.supports && relList.supports("modulepreload")) {
      return;
    }
    for (const link of document.querySelectorAll('link[rel="modulepreload"]')) {
      processPreload(link);
    }
    new MutationObserver((mutations) => {
      for (const mutation of mutations) {
        if (mutation.type !== "childList") {
          continue;
        }
        for (const node of mutation.addedNodes) {
          if (node.tagName === "LINK" && node.rel === "modulepreload")
            processPreload(node);
        }
      }
    }).observe(document, {
      childList: true,
      subtree: true
    });
    function getFetchOpts(script) {
      const fetchOpts = {};
      if (script.integrity)
        fetchOpts.integrity = script.integrity;
      if (script.referrerpolicy)
        fetchOpts.referrerPolicy = script.referrerpolicy;
      if (script.crossorigin === "use-credentials")
        fetchOpts.credentials = "include";
      else if (script.crossorigin === "anonymous")
        fetchOpts.credentials = "omit";
      else
        fetchOpts.credentials = "same-origin";
      return fetchOpts;
    }
    function processPreload(link) {
      if (link.ep)
        return;
      link.ep = true;
      const fetchOpts = getFetchOpts(link);
      fetch(link.href, fetchOpts);
    }
  })();
  const App_vue_vue_type_style_index_0_scoped_e0d9fbd5_lang = "";
  const _sfc_main = {
    components: {
      Content,
      Button: button,
      UnusedButton,
      Images
    },
    setup() {
      const count = ref(0);
      const inc = () => {
        count.value++;
      };
      return {
        count,
        inc
      };
    }
  };
  const _withScopeId = (n) => (pushScopeId("data-v-e0d9fbd5"), n = n(), popScopeId(), n);
  const _hoisted_1 = _withScopeId(() => createBaseVNode("h3", null, "Main App", -1));
  const _hoisted_2 = _withScopeId(() => createBaseVNode("hr", null, null, -1));
  function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_Images = resolveComponent("Images");
    const _component_Content = resolveComponent("Content");
    const _component_Button = resolveComponent("Button");
    const _component_UnusedButton = resolveComponent("UnusedButton");
    return openBlock(), createElementBlock("div", null, [
      _hoisted_1,
      createVNode(_component_Images),
      _hoisted_2,
      createVNode(_component_Content),
      createVNode(_component_Button),
      createVNode(_component_UnusedButton)
    ]);
  }
  const App = _export_sfc(_sfc_main, [
    [
      "render",
      _sfc_render
    ],
    [
      "__scopeId",
      "data-v-e0d9fbd5"
    ]
  ]);
  const app = createApp(App);
  app.use(createPinia());
  app.mount("#root");
});
```

**Comment by Menci** [comment]
Fixed in https://github.com/Menci/vite-plugin-top-level-await/commit/553fa88397f443756812f5acd07b4b56f1312204 and released in `v1.2.3`.

**Comment by dschewchenko** [comment]
Thank you very much! Will check this fix after some tasks =)
